best-practices/README.md

```diff
@@ -0,0 +1,75 @@
+
+Is that a good introduction for you to the bu-guide-to-git repository? Does it provide you wany information on the goal of the repo, whether it achieved it and whether it's usable or even in use?
+
+Of course not! That README.md just repeats the title of the repo with no additional information.
+
+### README Driven Development
+
+There are advocates of starting any development project by first creating your README.md file. This concept is called [README Driven Development](http://tom.preston-werner.com/2010/08/23/readme-driven-development.html). While we certainly would encourage you to try RDD, it's not necessary to having a great README or a great repo.
+
+However, there are a few basic questions that any good README.md should be able to answer.
+
+### Components of a Great README.md file
+
+Primarily, the README.md file should be a good starting point for a complete stranger figuring out a repo. Does your README answer these questions? If not, before writing another line of code update your README so that these questions are answered.
+
+- What does it do and why does it do it?
+- Where is it deployed?
+- How are changes deployed?
+- Can I run it locally and if so how?
+- How to test?
+- Oddities ( for instance homepage uses AWS SDK 2 instead of 3 because of PHP5 )
+- What configuration items are there | Is there an initial (one-time) setup?
+- Are any credentials needed?
+
+If your README can answer all these questions you likely have a great README!
+
+#### Examples of Real READMEs
+
+Below are a few examples of READMEs we think are stellar examples:
+
+[bu-infographics](https://github.com/bu-ist/bu-infographics/blob/develop/README.md)
+
+### Commented Out Code
+
+When making changes to a remote Git repository ( e.g. Github.com ), particularly `master`, or otherwise shared code don't persist commented out code.
+
+- [Don't commit commented out code](https://kentcdodds.com/blog/please-dont-commit-commented-out-code).
+- [Don't share commented out code](https://www.nayuki.io/page/dont-share-commented-out-code).
+
+Code you're `pushing` or otherwise sharing should not include commented out code. Doing so makes reading your code take longer for **no gain**.
+
+By all means comment out code locally while working out a change but when sharing it by publishing it to github.com it should only contain used code. If you wish to save commented out code remotely just in case do so in a branch explicitly marked for your personal use only. Remember to remove any remaning commented out code though before merging to `master` or other shared branches.
+
+### Branches are cheap
+
+Speaking of branches, in Git branches are cheap and easy to use. Almost every workflow at BU will benefit from taking advantage of branches. Many codebases at BU use [Gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) to varying degrees but nearly all codebases utilize branching to ensure all code in `master` is production ready.
```

Comment:
Not sure where, but should we recommend some graphical git clients in this readme? 

Although it may be worth cautioning not to rely on any one client too much, and that it is still a good idea to understand command line git.

Everybody has their own preference, but having a good graphical client can help a lot.

Some free suggestions may be:

- Sourcetree
- GitHub Desktop
- GitKraken

And then I think there's people that use Tower (although it is not free).

Reply:
Good suggestion but going to hold off as ideally someone more experience with GUI Git clients would write up a section for it.

Reply:
https://github.com/bu-ist/coding-standards/issues/45